### DISCUSSION TITLE
IV Infusion code

### DISCUSSION THREAD

**Discussion by LiupengWong** [nlmixr2/nlmixr2#261] [Q&A] [answered]
I am a novice in quantitative pharmacology and I'm learning to use nlmixr2 to solve modeling problems. I encountered an issue where the shrinkage value is 100% when running the model with the code below. I am not sure where the problem lies. Here is the code and the data header.

```
model.1cpt.ode <- function() {
  ini({
    tcl <- log(2.5)
    tv  <- log(15)

    eta.cl ~ 0.1
    eta.v ~ 0.11
    add.sd <- 0.1
  })
  model({
    cl <- exp(tcl + eta.cl)
    v <- exp(tv + eta.v)
    d/dt(depot) = -cl/v * depot
    cp = depot / v
    cp ~ add(add.sd)
  })
}
```

![CA57E2FC-09FE-4dac-9AF3-71EA628119C1.png](https://github.com/user-attachments/assets/eba1da93-eb57-4b46-84e9-c194f89dd127)

**Comment by LiupengWong** [comment]
sample size N=6.

**Comment by billdenney** [comment, marked as answer]
I think that the issue is with the dataset, but I'm going to give a couple of suggestions.

For the model, you've named your compartment receiving the IV dose "depot".  That compartment would typically be named "central".  It doesn't matter what you name it in this case (names can be whatever you want with the exception of `linCmt()` models).  But, it will be clearer to others if you use the name "central" as below:

```
model.1cpt.ode <- function() {
  ini({
    tcl <- log(2.5)
    tv  <- log(15)

    eta.cl ~ 0.1
    eta.v ~ 0.11
    add.sd <- 0.1
  })
  model({
    cl <- exp(tcl + eta.cl)
    v <- exp(tv + eta.v)
    d/dt(central) = -cl/v * central
    cp = central / v
    cp ~ add(add.sd)
  })
}
```

But the core issue is in the dataset, you've set CMT = 2, but you only have one compartment in your model.  I would suggest switching to named CMT in your data rather than numbered.  If you use the adjustment to the model above, you would set `CMT = "central"` in your dataset, and that will likely solve the issue.

**Reply by LiupengWong** [reply]
Thanks Bill, I updated according to your code, but it seems did not work.

It might still be an issue with the dataset. But it is  very tough for me to find the issue in the dataset.I uploaded the dataset; if it's convenient, could you please have a view? Thank you very much!
[modeldata.csv](https://github.com/user-attachments/files/16963827/modeldata.csv)

**Reply by john-harrold** [reply]
Hello Liupeng. I looked at the dataset and the CMT is still set to 2. I think Bill suggested you set it to 1 (your' first compartment) or the name of the compartment in the model. Did you  possibly upload the wrong dataset?

**Reply by LiupengWong** [reply]
Yes, I have double checked the dateset. I indeed did not update the value of CMT. So I undated the CMT from 1 to 2 and re-run the model. It worked. The model seems good !!!
Thank you John!

         Est.     SE %RSE Back-transformed(95%CI) BSV(CV%) Shrink(SD)%
tcl    -0.978 0.0541 5.53    0.376 (0.338, 0.418)     9.92      17.3% 
tv     0.0777 0.0721 92.8      1.08 (0.938, 1.24)     16.3     0.667% 
add.sd  0.269                               0.269
